### PR TITLE
Animation features

### PR DESCRIPTION
**Pull request by Chringo**
I'm working on it. Do not approve until I'm done with the conflicts. Please review this with utmost care for dev-branch.

### CODE REVIEW

**Comment by Chringo** [comment]
To many conflicts...

File: SSPSolution/GraphicsDLL/GraphicsHandler.cpp [hunk 1]
```diff
@@ -294,17 +408,23 @@ int GraphicsHandler::Render(float deltaTime)
 
 	m_shaderControl->SetActive(ShaderControl::Shaders::DEFERRED);
 	m_shaderControl->SetVariation(ShaderLib::ShaderVariations::Normal);
-
+	Resources::Model* modelPtr = nullptr;
```
Comment (Chringo) [inline SSPSolution/GraphicsDLL/GraphicsHandler.cpp]: This was removed in auto-merge. Is it suppose to be gone? Functions lower is dependent on this pointer.

File: SSPSolution/SSPSolution/LevelState.cpp [hunk 2]
```diff
@@ -1139,21 +1226,26 @@ int LevelState::CreateLevel(LevelData::Level * data)
...
 		else {
 			
 			DynamicEntity* tde = new DynamicEntity();
-			tde->SetGraphicsComponent(t_gc);
-			tde->SetPhysicsComponent(t_pc);
-			tde->Initialize(t_pc->PC_entityID, t_pc, t_gc);
+			//tde->SetGraphicsComponent(t_gc);
+			//tde->SetPhysicsComponent(t_pc);
+			//tde->SetAnimationComponent(t_anim);
+			tde->Initialize(t_pc->PC_entityID, t_pc, t_gc, t_anim);// Entity needs its ID
```
Comment (Chringo) [inline SSPSolution/SSPSolution/LevelState.cpp:1253]: Entities need to be initialised. This was a commit I did couple days ago. Recommitted as a fix.

File: SSPSolution/GraphicsDLL/AnimationHandler.cpp [hunk 3]
```diff
@@ -2,136 +2,181 @@
...
-	this->m_graphicsAnimationComponent->jointCount = 19;
-
-	this->m_graphicsAnimationComponent->worldMatrix = DirectX::XMMatrixTranslation(0, 4, 10);
-	for (int i = 0; i < 32; i++)
-	{
-		m_graphicsAnimationComponent->finalJointTransforms[i] = DirectX::XMMatrixIdentity();
-	}
-
-	m_TransitionTimeLeft = 0;
-	m_TransitionDuration = 0;
```
Comment (MartinClementson) [inline SSPSolution/GraphicsDLL/AnimationHandler.cpp]: Why is the whole constructor gone?

File: SSPSolution/GraphicsDLL/GraphicsHandler.cpp (hunk 1 above)
Comment (Marcets423) [reply SSPSolution/GraphicsDLL/GraphicsHandler.cpp]: Yes, It is supposed to be gone.

File: SSPSolution/ResourceLib/ModelHandler.cpp [hunk 4]
```diff
@@ -239,6 +233,14 @@ void Resources::ModelHandler::SetDevice(ID3D11Device * device)
  m_meshHandler->SetDevice(device);
  m_materialHandler->SetDevice(device);
  m_skeletonHandler->SetDevice(device);
+
+ ResourceContainer* temp;
+ unsigned int one = 2759249725;
+ unsigned int two = 3255160373;
+ FileLoader::GetInstance()->OpenFile(FileLoader::BPF_FILE);
```
Comment (Marcets423) [inline SSPSolution/ResourceLib/ModelHandler.cpp:240]: This is not a good idea. I suggest we open an issue on how to solve the loading of the character models

**Review by JohnAugustsson — APPROVED** [review]
Looks fine

**Pull request merged**